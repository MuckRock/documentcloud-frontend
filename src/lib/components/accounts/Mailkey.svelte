<script lang="ts">
  import {_} from 'svelte-i18n';
  import Button from "../common/Button.svelte";
  import Flex from "../common/Flex.svelte";
  import Modal from "../layouts/Modal.svelte";
  import { createMailkey, destroyMailkey } from '$lib/api/accounts';

  export let fetch = globalThis.fetch;
  export let message: string = '';
  let error = false;

  function reset() {
    message = '';
    error = false;
  }

  async function create() {
    reset();
    const mailkey = await createMailkey(fetch);
    if (mailkey) {
      message = $_("mailkey.create.success", {
        values: { mailkey: mailkey },
      });
    } else {
      error = true;
      message = $_("mailkey.create.failure")
    }
  }

  async function destroy() {
    reset();
    if (await destroyMailkey(fetch)) {
      message = $_("mailkey.destroy.success");
    } else {
      error = true;
      message = $_("mailkey.destroy.failure");
    }
  }
</script>

<Modal on:close>
  <h1 slot="title">{$_("mailkey.title")}</h1>
  <div class="description">{@html $_("mailkey.description")}</div>
  {#if message}
    <p class:error>
      {@html message}
    </p>
  {/if}
  <Flex gap={1} wrap justify="center">
    <Button mode="primary" on:click={create}>{$_("mailkey.create.button")}</Button>
    <Button mode="danger" on:click={destroy}>{$_("mailkey.destroy.button")}</Button>
  </Flex>
</Modal>

<style>
  .description {
    columns: 2;
    column-gap: 2rem;
    margin-bottom: 1rem;
    font-size: var(--font-sm);
  }
  :global(.description p) {
    margin-bottom: 1rem;
    -webkit-column-break-inside: avoid;
  }
  .error {
    color: var(--red-3);
  }
  @media (max-width: 42rem) {
    .description {
      columns: 1;
    }
  }
</style>