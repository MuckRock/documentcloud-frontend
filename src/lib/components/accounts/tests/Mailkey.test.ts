import { test, expect } from "vitest";
import { render } from "@testing-library/svelte";
import Mailkey from "../Mailkey.svelte";

test("Mailkey", () => {
  let result = render(Mailkey);
  expect(result.container).toMatchSnapshot();
  result = render(Mailkey, {message: 'Something happened'});
  expect(result.container).toMatchSnapshot();
});
