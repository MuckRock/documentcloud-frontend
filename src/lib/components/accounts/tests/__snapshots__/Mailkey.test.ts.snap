// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Mailkey 1`] = `
<div>
  <div
    class="backdrop svelte-65lbzd"
  >
    <div
      class="dialog card svelte-65lbzd"
    >
      <header
        class="svelte-65lbzd"
      >
        <button
          class="primary normal ghost svelte-12ugz1l"
          title=""
          type="button"
        >
          <svg
            class="octicon octicon-x-circle"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.036 7.976a.75.75 0 0 0-1.06 1.06L10.939 12l-2.963 2.963a.75.75 0 1 0 1.06 1.06L12 13.06l2.963 2.964a.75.75 0 0 0 1.061-1.06L13.061 12l2.963-2.964a.75.75 0 1 0-1.06-1.06L12 10.939 9.036 7.976Z"
            />
            <path
              d="M12 1c6.075 0 11 4.925 11 11s-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1ZM2.5 12a9.5 9.5 0 0 0 9.5 9.5 9.5 9.5 0 0 0 9.5-9.5A9.5 9.5 0 0 0 12 2.5 9.5 9.5 0 0 0 2.5 12Z"
            />
          </svg>
        </button>
        
         
        <h1
          slot="title"
        >
          Upload via Email
        </h1>
      </header>
       
      <main
        class="content svelte-65lbzd"
      >
        <div
          class="description svelte-mrguwb"
        >
          <p>
            You can upload documents to your account by sending them to to a special email address as attachments.
          </p>
          <p>
            <strong>
              For security reasons, this email is only shown to you once. Please copy it to a secure location.
            </strong>
          </p>
          <p>
            This address will accept attachments from any email account. Documents are uploaded as private. Generating a new upload address will disable any previously created addresses.
          </p>
          <p>
            We’d love your feedback and to hear about creative use cases at 
            <a
              href="mailto:info@documentcloud.org"
              target="_blank"
            >
              info@documentcloud.org
            </a>
            .
          </p>
        </div>
         
         
        <div
          style="display: flex; flex-direction: row; flex-wrap: wrap; align-items: stretch; justify-content: center; gap: 1rem;"
        >
          <button
            class="primary normal minW svelte-12ugz1l"
            title=""
            type="button"
          >
            Create New Email Address
          </button>
          
           
          <button
            class="danger normal minW svelte-12ugz1l"
            title=""
            type="button"
          >
            Delete Existing Address
          </button>
          
        </div>
      </main>
    </div>
  </div>
</div>
`;

exports[`Mailkey 2`] = `
<div>
  <div
    class="backdrop svelte-65lbzd"
  >
    <div
      class="dialog card svelte-65lbzd"
    >
      <header
        class="svelte-65lbzd"
      >
        <button
          class="primary normal ghost svelte-12ugz1l"
          title=""
          type="button"
        >
          <svg
            class="octicon octicon-x-circle"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.036 7.976a.75.75 0 0 0-1.06 1.06L10.939 12l-2.963 2.963a.75.75 0 1 0 1.06 1.06L12 13.06l2.963 2.964a.75.75 0 0 0 1.061-1.06L13.061 12l2.963-2.964a.75.75 0 1 0-1.06-1.06L12 10.939 9.036 7.976Z"
            />
            <path
              d="M12 1c6.075 0 11 4.925 11 11s-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1ZM2.5 12a9.5 9.5 0 0 0 9.5 9.5 9.5 9.5 0 0 0 9.5-9.5A9.5 9.5 0 0 0 12 2.5 9.5 9.5 0 0 0 2.5 12Z"
            />
          </svg>
        </button>
        
         
        <h1
          slot="title"
        >
          Upload via Email
        </h1>
      </header>
       
      <main
        class="content svelte-65lbzd"
      >
        <div
          class="description svelte-mrguwb"
        >
          <p>
            You can upload documents to your account by sending them to to a special email address as attachments.
          </p>
          <p>
            <strong>
              For security reasons, this email is only shown to you once. Please copy it to a secure location.
            </strong>
          </p>
          <p>
            This address will accept attachments from any email account. Documents are uploaded as private. Generating a new upload address will disable any previously created addresses.
          </p>
          <p>
            We’d love your feedback and to hear about creative use cases at 
            <a
              href="mailto:info@documentcloud.org"
              target="_blank"
            >
              info@documentcloud.org
            </a>
            .
          </p>
        </div>
         
        <p
          class="message svelte-mrguwb"
        >
          Something happened
        </p>
         
        <div
          style="display: flex; flex-direction: row; flex-wrap: wrap; align-items: stretch; justify-content: center; gap: 1rem;"
        >
          <button
            class="primary normal minW svelte-12ugz1l"
            title=""
            type="button"
          >
            Create New Email Address
          </button>
          
           
          <button
            class="danger normal minW svelte-12ugz1l"
            title=""
            type="button"
          >
            Delete Existing Address
          </button>
          
        </div>
      </main>
    </div>
  </div>
</div>
`;
